src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -0,0 +1,671 @@
+        Buffer + PayloadLengthOffset);
+    *HeaderLength = *PacketLength;
+
+    CxPlatCopyMemory(Buffer + *PacketLength, CryptoBuffer, CryptoBufferLength);
+    *PacketLength += CryptoBufferLength;
+    *PacketLength += CXPLAT_ENCRYPTION_OVERHEAD;
+}
+
+
+void fuzzInitialPacket(CXPLAT_SOCKET* Binding, CXPLAT_ROUTE Route){
+    const StrBuffer InitialSalt("afbfec289993d24c9e9786f19c6111e04390a899");
+    const uint16_t DatagramLength = QUIC_MIN_INITIAL_LENGTH; 
+    uint64_t StartTimeMs = CxPlatTimeMs64();
+    while (CxPlatTimeDiff64(StartTimeMs, CxPlatTimeMs64()) < RunTimeMs) {
+        CXPLAT_SEND_CONFIG SendConfig = { &Route, DatagramLength, CXPLAT_ECN_NON_ECT, 0 };
+        CXPLAT_SEND_DATA* SendData = CxPlatSendDataAlloc(Binding, &SendConfig);
+        if(!SendData){
+            printf("CxPlatSendDataAlloc failed\n");
+        }
+        while (CxPlatTimeDiff64(StartTimeMs, CxPlatTimeMs64()) < RunTimeMs && !CxPlatSendDataIsFull(SendData)) {
+            const uint64_t PacketNumber = 0; //fuzz
+            uint8_t Packet[512] = {0};
+            uint16_t PacketLength, HeaderLength;
+
+            WriteClientInitialPacket(
+                PacketNumber,
+                sizeof(uint64_t),
+                sizeof(Packet),
+                Packet,
+                &PacketLength,
+                &HeaderLength);
+
+            uint8_t *PacketBuffer = Packet;
+            QUIC_LONG_HEADER_V1* Header = (QUIC_LONG_HEADER_V1*)Packet;
+            Header->IsLongHeader = GetRandom(2);
+            Header->FixedBit = GetRandom(2);
+            Header->Reserved = GetRandom(2);
+            Header->PnLength = GetRandom(sizeof(uint32_t));
+            Header->DestCidLength  = (uint8_t)GetRandom(sizeof(uint64_t));
+
+            PacketBuffer += sizeof(QUIC_LONG_HEADER_V1) + sizeof(uint64_t); // point to the source id length
+            *PacketBuffer = (uint8_t)GetRandom(sizeof(uint64_t)); // fuzz the source id length
+
+            uint16_t PacketNumberOffset = HeaderLength - sizeof(uint32_t);
+
+            uint64_t* DestCid = (uint64_t*)(Packet + sizeof(QUIC_LONG_HEADER_V1));
+            uint64_t* SrcCid = (uint64_t*)(Packet + sizeof(QUIC_LONG_HEADER_V1) + sizeof(uint64_t) + sizeof(uint8_t));
+
+            uint64_t* OrigSrcCid = nullptr;
+            for (uint16_t i = HeaderLength; i < PacketLength; ++i) {
+                if (MagicCid == *(uint64_t*)&Packet[i]) {
+                    OrigSrcCid = (uint64_t*)&Packet[i];
+                }
+            }
+            if (!OrigSrcCid) {
+                printf("Failed to find OrigSrcCid!\n");
+                return;
+            }
+
+            CxPlatRandom(sizeof(uint64_t), DestCid); //fuzz
+            CxPlatRandom(sizeof(uint64_t), SrcCid); //fuzz
+            QUIC_BUFFER* SendBuffer =
+                CxPlatSendDataAllocBuffer(SendData, DatagramLength);
+             if(!SendBuffer) {
+                printf("CxPlatSendDataAllocBuffer failed\n");
+                }
```

Comment:
```suggestion
            QUIC_BUFFER* SendBuffer =
                CxPlatSendDataAllocBuffer(SendData, DatagramLength);
            if (!SendBuffer) {
                printf("CxPlatSendDataAllocBuffer failed\n");
            }
```
Also, if this fails, shouldn't we return?